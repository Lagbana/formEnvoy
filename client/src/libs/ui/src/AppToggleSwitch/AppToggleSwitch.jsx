import React from 'react'
import { SwitchToggle } from 'dooboo-ui'

import { Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles'

export const AppToggleSwitch = ({ enabled = false, ...props }) => {
  const styles = {
    container: {
      width: 50,
      height: 26,
      padding: 4,
      borderRadius: 13
    },
    circle: {
      width: 18,
      height: 18,
      borderRadius: 9
    }
  }

  return (
    <SwitchToggle
      duration={200}
      switchOn={enabled}
      containerStyle={styles.container}
      circleStyle={styles.circle}
      circleColorOn={Colors.white}
      backgroundColorOff='#e0e0e0'
      backgroundColorOn={Colors.green}
      onPress={() => {}}
      {...props}
    />
  )
}
