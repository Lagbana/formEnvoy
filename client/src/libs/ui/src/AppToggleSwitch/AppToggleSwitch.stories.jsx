import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppToggleSwitch } from './AppToggleSwitch'

storiesOf('AppToggleSwitch', module)
  .add('Enabled: True', () => <AppToggleSwitch enabled={true} />)
  .add('Enabled: False', () => <AppToggleSwitch enabled={false} />)
