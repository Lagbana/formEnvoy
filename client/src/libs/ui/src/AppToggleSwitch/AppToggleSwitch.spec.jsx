import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppToggleSwitch } from './AppToggleSwitch'
configure({ adapter: new Adapter() })

jest.mock('@dooboo-ui/native-switch-toggle', () => 'switch-toggle')

test('<AppToggleSwitch /> renders correctly', () => {
  const component = shallow(<AppToggleSwitch />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
