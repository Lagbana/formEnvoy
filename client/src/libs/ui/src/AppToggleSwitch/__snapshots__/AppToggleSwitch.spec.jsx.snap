// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppToggleSwitch /> renders correctly 1`] = `""`;
